samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java

```diff
@@ -2319,6 +2371,560 @@ public String getOwner() {
+						}
+						if (data != null) {
+							if (data instanceof Double) {
+								cell.setCellValue(((Double)data).doubleValue());
+							} else {
+								// stripping html for export, SAK-17021
+								cell.setCellValue(data.toString());
+
+							}
+						}
+					}
+				}
+			}
+			
+		}
+		
+		return wb;
+	}
+
+	private int findColumnSize(List<List<Object>> spreadsheetData) {
+        int columns = 0; // the largest number of columns required for a row
+	    for (List<Object> list : spreadsheetData) {
+            if (list != null && list.size() > columns) {
+                columns = list.size();
+            }
+        }
+        return columns;
+    }
+	
+	private Cell createCell(Row row, short column, CellStyle cellStyle) {
+		Cell cell = row.createCell(column);
+		//cell.setEncoding(HSSFCell.ENCODING_UTF_16);	
+		if (cellStyle != null) {
+			cell.setCellStyle(cellStyle);
+		}
+		
+		return cell;
+	}
+	
+  	public List getExportResponsesData(String poolId) {
+  		ArrayList dataList = new ArrayList();
+  		ArrayList headerList = new ArrayList();
+  		ArrayList finalList = new ArrayList(2);
+  	  
+  		Float itemScore = null;
+  		
+  		String questionString = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_question");
+        String responseString = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_response");
+        String feedbackString = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_feedback");
+        
+  		// Create Header List
+  		headerList.add("");
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_text"));
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_type"));
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_points"));
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_discount"));
+  		headerList.add(responseString.concat(" A"));
+  		headerList.add(feedbackString.concat(" A"));
+  		headerList.add(responseString.concat(" B"));
+  		headerList.add(feedbackString.concat(" B"));
+  		headerList.add(responseString.concat(" C"));
+  		headerList.add(feedbackString.concat(" C"));
+  		headerList.add(responseString.concat(" D"));
+  		headerList.add(feedbackString.concat(" D"));
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_key"));	  
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_feedbackCorrect"));
+  		headerList.add(ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.QuestionPoolMessages","q_feedbackIncorrect"));
+
+  		int maxAnswers = 4;
```

Comment:
What is `4` in this context? This should probably be a constant with descriptive name and/or comment